src/data_structures/qgram_index.rs

```diff
@@ -108,6 +108,11 @@ impl QGramIndex {
         &self.pos[self.address[qgram]..self.address[qgram + 1]]
     }
 
+    /// Iterate over the q-gram ranks of pattern
```

Comment:
I don't think this method belongs into the qgram index. As you don't need it anyways, would it be ok to remove it?